streams-contrib/streams-provider-sprinklr/src/main/java/org/apache/streams/sprinklr/util/SprinklrDataToActivityConverter.java

```diff
@@ -0,0 +1,251 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.streams.sprinklr.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.commons.lang.StringUtils;
+import org.apache.streams.pojo.extensions.ExtensionUtil;
+import org.apache.streams.pojo.json.*;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.List;
+
+import static org.apache.streams.data.util.ActivityUtil.*;
+
+/**
+ * Converts SprinklrData to an Activity Object
+ */
+public class SprinklrDataToActivityConverter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SprinklrDataToActivityConverter.class);
+    public static final String LANGUAGE_KEY = "LANGUAGE";
+
+    /**
+     * convert the given Sprinklr data in JsonNode format to an Activity
+     * @param item JsonNode
+     * @return an Activity
+     */
+    public Activity convert(JsonNode item) {
+        Activity activity = new Activity();
+
+        try {
+            activity.setVerb("post");
+            activity.setId(formatId(activity.getVerb(), item.get("universalMessageId").asText()));
+            activity.setContent(item.has("message") ? item.get("message").asText() : "");
+            activity.setTitle(item.has("title") ? item.get("title").asText() : "");
+            activity.setPublished(item.has("createdTime") ? new DateTime(item.get("createdTime").asLong()) : new DateTime());
+            activity.setUrl(item.has("permalink") ? item.get("permalink").asText() : "");
+            activity.setActor(buildActor(item));
+
+            // set media data
+            activity.setObject(buildActivityObject(item));
+            activity.setProvider(getProvider(item));
+
+            Map<String, Object> extensions = ExtensionUtil.ensureExtensions(activity);
+            extensions.put("sprinkler", item);
+
+            // set location, set language
+            setActivityExtensions(item, extensions);
+        } catch (Exception e) {
+            LOGGER.error("Failed to convert JsonNode to Activity; errorMessage={}", e.getMessage());
+        }
+
+        return activity;
+    }
+
+    /**
+     * Set Activity Extension properties for location and language
+     * @param item JsonNode
+     * @param extensions a Map of String to Object
+     */
+    @SuppressWarnings("unchecked")
+    private void setActivityExtensions(JsonNode item, Map<String, Object> extensions) {
+        Map<String, Object> locationMap;
```

Comment:
Sprinklr returns different data for each social network. I have gone through the provided documentation and updated the mapping to include more details of the activity.